Warnings: "... specifies less restrictive attributes than its target"

Hi there,

I'm trying to integrate rpmalloc in my program and I get a lot of warnings when compiling with gcc 11.3.0:

```
[1/4] Building C object src/CMakeFiles/rpmalloc.dir/external/rpmalloc/rpmalloc/rpmalloc.c.o
In file included from rpmalloc.c:3561:
malloc.c:466:7: warning: ‘__libc_memalign’ specifies less restrictive attributes than its target ‘rpmemalign’: ‘alloc_size’, ‘malloc’ [-Wmissing-attributes]
  466 | void* __libc_memalign(size_t align, size_t size) RPALIAS(rpmemalign)
      |       ^~~~~~~~~~~~~~~
rpmalloc.c:3145:1: note: ‘__libc_memalign’ target declared here
 3145 | rpmemalign(size_t alignment, size_t size) {
      | ^~~~~~~~~~
In file included from rpmalloc.c:3561:
malloc.c:463:7: warning: ‘__libc_realloc’ specifies less restrictive attributes than its target ‘rprealloc’: ‘alloc_size’, ‘malloc’ [-Wmissing-attributes]
  463 | void* __libc_realloc(void* p, size_t size) RPALIAS(rprealloc)
      |       ^~~~~~~~~~~~~~
rpmalloc.c:3088:1: note: ‘__libc_realloc’ target declared here
 3088 | rprealloc(void* ptr, size_t size) {
      | ^~~~~~~~~
In file included from rpmalloc.c:3561:
malloc.c:462:7: warning: ‘__libc_calloc’ specifies less restrictive attributes than its target ‘rpcalloc’: ‘alloc_size’, ‘malloc’ [-Wmissing-attributes]
  462 | void* __libc_calloc(size_t count, size_t size) RPALIAS(rpcalloc)
      |       ^~~~~~~~~~~~~
rpmalloc.c:3061:1: note: ‘__libc_calloc’ target declared here
 3061 | rpcalloc(size_t num, size_t size) {
      | ^~~~~~~~
In file included from rpmalloc.c:3561:
malloc.c:461:7: warning: ‘__libc_malloc’ specifies less restrictive attributes than its target ‘rpmalloc’: ‘alloc_size’, ‘malloc’ [-Wmissing-attributes]
  461 | void* __libc_malloc(size_t size) RPALIAS(rpmalloc)
      |       ^~~~~~~~~~~~~
rpmalloc.c:3044:1: note: ‘__libc_malloc’ target declared here
 3044 | rpmalloc(size_t size) {
      | ^~~~~~~~
In file included from rpmalloc.c:3561:
malloc.c:249:7: warning: ‘memalign’ specifies less restrictive attributes than its target ‘rpmemalign’: ‘alloc_size’, ‘malloc’ [-Wmissing-attributes]
  249 | void* memalign(size_t alignment, size_t size) RPALIAS(rpmemalign)
      |       ^~~~~~~~
rpmalloc.c:3145:1: note: ‘memalign’ target declared here
 3145 | rpmemalign(size_t alignment, size_t size) {
      | ^~~~~~~~~~
In file included from rpmalloc.c:3561:
malloc.c:247:7: warning: ‘reallocf’ specifies less restrictive attributes than its target ‘rprealloc’: ‘alloc_size’, ‘malloc’ [-Wmissing-attributes]
  247 | void* reallocf(void* ptr, size_t size) RPALIAS(rprealloc)
      |       ^~~~~~~~
rpmalloc.c:3088:1: note: ‘reallocf’ target declared here
 3088 | rprealloc(void* ptr, size_t size) {
      | ^~~~~~~~~
In file included from rpmalloc.c:3561:
malloc.c:207:7: warning: ‘_Znam’ specifies less restrictive attributes than its target ‘rpmalloc’: ‘alloc_size’, ‘malloc’ [-Wmissing-attributes]
  207 | void* _Znam(uint64_t size) RPALIAS(rpmalloc)
      |       ^~~~~
rpmalloc.c:3044:1: note: ‘_Znam’ target declared here
 3044 | rpmalloc(size_t size) {
      | ^~~~~~~~
In file included from rpmalloc.c:3561:
malloc.c:206:7: warning: ‘_Znwm’ specifies less restrictive attributes than its target ‘rpmalloc’: ‘alloc_size’, ‘malloc’ [-Wmissing-attributes]
  206 | void* _Znwm(uint64_t size) RPALIAS(rpmalloc)
      |       ^~~~~
rpmalloc.c:3044:1: note: ‘_Znwm’ target declared here
 3044 | rpmalloc(size_t size) {
      | ^~~~~~~~
```

--- a/rpmalloc/malloc.c
+++ b/rpmalloc/malloc.c
@@ -203,8 +203,8 @@ void _ZdaPv(void* p) RPALIAS(rpfree)
 
 #if ARCH_64BIT
 // 64-bit operators new and new[], normal and aligned
-void* _Znwm(uint64_t size) RPALIAS(rpmalloc)
-void* _Znam(uint64_t size) RPALIAS(rpmalloc)
+void* _Znwm(uint64_t size) RPMALLOC_ATTRIB_MALLOC RPMALLOC_ATTRIB_ALLOC_SIZE(1) RPALIAS(rpmalloc)
+void* _Znam(uint64_t size) RPMALLOC_ATTRIB_MALLOC RPMALLOC_ATTRIB_ALLOC_SIZE(1) RPALIAS(rpmalloc)
 void* _Znwmm(uint64_t size, uint64_t align) RPALIAS(rpaligned_alloc_reverse)
 void* _Znamm(uint64_t size, uint64_t align) RPALIAS(rpaligned_alloc_reverse)
 void* _ZnwmSt11align_val_t(size_t size, size_t align) RPALIAS(rpaligned_alloc_reverse)
@@ -222,8 +222,8 @@ void _ZdlPvmSt11align_val_t(void* p, size_t n, size_t a) RPALIAS(rpfree_size_ali
 void _ZdaPvmSt11align_val_t(void* p, size_t n, size_t a) RPALIAS(rpfree_size_aligned)
 #else
 // 32-bit operators new and new[], normal and aligned
-void* _Znwj(uint32_t size) RPALIAS(rpmalloc)
-void* _Znaj(uint32_t size) RPALIAS(rpmalloc)
+void* _Znwj(uint32_t size) RPMALLOC_ATTRIB_MALLOC RPMALLOC_ATTRIB_ALLOC_SIZE(1) RPALIAS(rpmalloc)
+void* _Znaj(uint32_t size) RPMALLOC_ATTRIB_MALLOC RPMALLOC_ATTRIB_ALLOC_SIZE(1) RPALIAS(rpmalloc)
 void* _Znwjj(uint32_t size, uint32_t align) RPALIAS(rpaligned_alloc_reverse)
 void* _Znajj(uint32_t size, uint32_t align) RPALIAS(rpaligned_alloc_reverse)
 void* _ZnwjSt11align_val_t(size_t size, size_t align) RPALIAS(rpaligned_alloc_reverse)
@@ -241,12 +241,12 @@ void _ZdlPvjSt11align_val_t(void* p, size_t n, size_t a) RPALIAS(rpfree_size_ali
 void _ZdaPvjSt11align_val_t(void* p, size_t n, size_t a) RPALIAS(rpfree_size_aligned)
 #endif
 
-void* malloc(size_t size) RPALIAS(rpmalloc)
+void* malloc(size_t size) RPMALLOC_ATTRIB_MALLOC RPMALLOC_ATTRIB_ALLOC_SIZE(1) RPALIAS(rpmalloc)
 void* calloc(size_t count, size_t size) RPALIAS(rpcalloc)
 void* realloc(void* ptr, size_t size) RPALIAS(rprealloc)
-void* reallocf(void* ptr, size_t size) RPALIAS(rprealloc)
+void* reallocf(void* ptr, size_t size) RPMALLOC_ATTRIB_MALLOC RPMALLOC_ATTRIB_ALLOC_SIZE(2) RPALIAS(rprealloc)
 void* aligned_alloc(size_t alignment, size_t size) RPALIAS(rpaligned_alloc)
-void* memalign(size_t alignment, size_t size) RPALIAS(rpmemalign)
+void* memalign(size_t alignment, size_t size) RPMALLOC_ATTRIB_MALLOC RPMALLOC_ATTRIB_ALLOC_SIZE(2) RPALIAS(rpmemalign)
 int posix_memalign(void** memptr, size_t alignment, size_t size) RPALIAS(rpposix_memalign)
 void free(void* ptr) RPALIAS(rpfree)
 void cfree(void* ptr) RPALIAS(rpfree)
@@ -458,12 +458,12 @@ pthread_create(pthread_t* thread,
 
 #if defined(__GLIBC__) && defined(__linux__)
 
-void* __libc_malloc(size_t size) RPALIAS(rpmalloc)
-void* __libc_calloc(size_t count, size_t size) RPALIAS(rpcalloc)
-void* __libc_realloc(void* p, size_t size) RPALIAS(rprealloc)
+void* __libc_malloc(size_t size) RPMALLOC_ATTRIB_MALLOC RPMALLOC_ATTRIB_ALLOC_SIZE(1) RPALIAS(rpmalloc)
+void* __libc_calloc(size_t count, size_t size) RPMALLOC_ATTRIB_MALLOC RPMALLOC_ATTRIB_ALLOC_SIZE2(1, 2) RPALIAS(rpcalloc)
+void* __libc_realloc(void* p, size_t size) RPMALLOC_ATTRIB_MALLOC RPMALLOC_ATTRIB_ALLOC_SIZE(2) RPALIAS(rprealloc)
 void __libc_free(void* p) RPALIAS(rpfree)
 void __libc_cfree(void* p) RPALIAS(rpfree)
-void* __libc_memalign(size_t align, size_t size) RPALIAS(rpmemalign)
+void* __libc_memalign(size_t align, size_t size) RPMALLOC_ATTRIB_MALLOC RPMALLOC_ATTRIB_ALLOC_SIZE(2) RPALIAS(rpmemalign)
 int __posix_memalign(void** p, size_t align, size_t size) RPALIAS(rpposix_memalign)
 
 extern void* __libc_valloc(size_t size);

